src/Microsoft.AspNetCore.Mvc.Analyzers/AvoidHtmlPartialAnalyzer.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.AspNetCore.Mvc.Analyzers
+{
+    public class AvoidHtmlPartialAnalyzer : ViewFeatureAnalyzerBase
+    {
+        public AvoidHtmlPartialAnalyzer()
+            : base(DiagnosticDescriptors.MVC1000_HtmlHelperPartialShouldBeAvoided)
+        {
+        }
+
+        protected override void InitializeWorker(ViewFeaturesAnalyzerContext analyzerContext)
```

Comment:
Do you need two analyzer classes for this? Can it be merged into one and just produce different error codes.

Reply:
Merged the two